packages/context-editor/src/components/AttributePanelToggleWidget.tsx

```diff
@@ -0,0 +1,102 @@
+import type { Node } from "prosemirror-model";
+import type { EditorView } from "prosemirror-view";
+
+import React from "react";
+import { useWidgetViewContext } from "@prosemirror-adapter/react";
+import { TextSelection } from "prosemirror-state";
+
+import { attributePanelKey } from "../plugins/attributePanel";
+import { reactPropsKey } from "../plugins/reactProps";
+
+export const AttributePanelToggleWidget = () => {
+	const { view, getPos, spec } = useWidgetViewContext();
+	if (!spec || !("node" in spec)) {
+		return null;
+	}
+	const attributePanelPluginState = attributePanelKey.getState(view.state);
+	if (!attributePanelPluginState) {
+		return null;
+	}
+	const { panelPosition, setPanelPosition } = attributePanelPluginState;
+
+	const onClick = () => {
+		const pos = getPos();
+		if (pos === undefined) {
+			return;
+		}
+		view.dispatch(
+			view.state.tr.setSelection(
+				new TextSelection(
+					view.state.doc.resolve(pos)
+					// view.state.doc.resolve(getPos() + node.nodeSize)
+				)
+			)
+		);
```

Comment:
this was from an attempt to make the attribute panel work based on the selection, rather than manually setting the `node` in the panelProps. it's not really working at the moment so you'll probably want to remove this